linkml/utils/generator.py

```diff
@@ -153,13 +154,17 @@ class Generator(metaclass=abc.ABCMeta):
     metamodel: SchemaLoader = None
     """A SchemaLoader instance that points to the LinkML metamodel (meta.yaml)"""
 
+    stacktrace: bool = False
+    """True means print stack trace, false just error message"""
+
     def __post_init__(self) -> None:
         if not self.logger:
             self.logger = logging.getLogger()
         #    logging.basicConfig()
         #    self.logger = logging.getLogger(self.__class__.__name__)
         #    self.logger.setLevel(log_level)
-
+        if not self.stacktrace:
+            sys.tracebacklimit = 0
```

Comment:
I don't think the generator object should be messing with globals

Instead this should be controlled entirely at CLI level

See https://github.com/linkml/linkml/issues/1227